source/common/upstream/cds_api_impl.cc

```diff
@@ -54,7 +71,8 @@ void CdsApiImpl::onConfigUpdate(const ResourceVector& resources, const std::stri
   for (auto& cluster : resources) {
     const std::string cluster_name = cluster.name();
     clusters_to_remove.erase(cluster_name);
-    if (cm_.addOrUpdateCluster(cluster, version_info)) {
+    if (cm_.addOrUpdateCluster(cluster, version_info,
+                               [this](auto, auto) { resumeCdsAfterWarming(); })) {
```

Comment:
Should there be some assertions around only being called with warmingClusterCount() == 0 if Finished?

Reply:
Added assertion. Also added a check to call resume only on the finished warming.